examples/timestepping/acoustic_superstep_2d.py

```diff
@@ -0,0 +1,252 @@
+        time_order=2, space_order=4
+    )
+    source.data[:, 0] = rick
+    src_term = source.inject(field=u.forward, expr=source*velocity*velocity*dt*dt)
+
+    op1 = Operator([stencil] + src_term)
+    op1(time=tn1 - 1, dt=dt)
+
+    # Stencil and operator
+    idx = tn1 % 3
+    if step == 1:
+        # Non-superstep case
+        new_u = TimeFunction(name="new_u", grid=grid, time_order=2, space_order=2)
+        stencil = [stencil.subs(
+            {u.forward: new_u.forward, u: new_u, u.backward: new_u.backward}
+        )]
+        new_u.data[0, :] = u.data[idx - 2]
+        new_u.data[1, :] = u.data[idx - 1]
+        new_u.data[2, :] = u.data[idx]
+    else:
+        new_u, new_u_p, *stencil = superstep_generator(u, stencil.rhs, step)
+
+        new_u.data[0, :] = u.data[idx - 1]
+        new_u.data[1, :] = u.data[idx]
+        new_u_p.data[0, :] = u.data[idx - 2]
+        new_u_p.data[1, :] = u.data[idx - 1]
+
+    tn2 = int(np.ceil((t2 - t1)/model.critical_dt))
+    dt = (t2 - t1)/tn2
+
+    # Snapshot the solution
+    factor = int(np.ceil(tn2/(snapshots + 1)))
+    t_sub = ConditionalDimension('t_sub', parent=grid.time_dim, factor=factor)
+    u_save = TimeFunction(
+        name='usave', grid=grid,
+        time_order=0, space_order=2,
+        save=snapshots//step + 1, time_dim=t_sub
+    )
+    save = Eq(u_save, new_u)
+
+    op = Operator([*stencil, save])
+    op(dt=dt)
+
+    if step == 1:
+        u_save.data[0, :, :] = u.data[idx]
+
+    return u_save.data
+
+
+layered_model = model(
+    name='layered',
+    velocity=layered_velocity,
+    shape=(101, 101),
+    origin=(0., 0.),
+    extent=(1000, 1000),  # 1kmx1km
+    source=(500, 20),
+    t0=0,
+    t1=0.2,
+    t2=0.65,
+    critical_dt=0.002357,
+    zlim=30
+)
+
+marmousi_model = model(
+    name='marmousi',
+    velocity=marmousi,
+    shape=(274, 301),
+    origin=(4875., 262.5),
+    extent=(3000, 2737.5),  # 3kmx2.7km
+    source=(1000, 1200),
+    t0=0,
+    t1=0.2,
+    t2=0.5,
+    critical_dt=0.0013728,
+    zlim=20
+)
+
+
+if __name__ == '__main__':
+    parser = ArgumentParser()
+    parser.add_argument('--model', default='layered', choices=['layered', 'marmousi'])
+    args = parser.parse_args()
+
+    model = layered_model if args.model == 'layered' else marmousi_model
+
+    # Supersteps
+    k = [1, 4]
+    # Snapshots
+    m = 13
+    fig, axes = plt.subplots(len(k), m)
+
+    # Velocity model
+    grid = Grid(shape=model.shape, extent=model.extent)
+    v = model.velocity(grid)
+    plot_extent = [
+        model.origin[0],
+        model.origin[0] + model.extent[0],
+        model.origin[1] + model.extent[1],
+        model.origin[1]
+    ]
+
+    for step, ax_row in zip(k, axes, strict=True):
+        data = acoustic_model(model, step=step, snapshots=m)
+        time = np.linspace(model.t1, model.t2, (m - 1)//step + 1)
+        idx = 0
+        for ii, ax in enumerate(ax_row):
```

Comment:
I would set up a separate plotting function for the contents of this loop to improve readability